README.md

```diff
@@ -1 +1,241 @@
+
+```go
+
+type JobRunner struct {
+  jobs map[int64]*Job
+  jobsByClient map[string]*Job
+  mu sync.Mutex
+}
+
+func (j *JobRunner) Start(command, client string) *JobRunner {
+  j.mu.Lock()
+  defer j.mu.Unlock()
+
+  // do stuff
+}
+```
+
+### CLI
+
+```text
+pjw list          Provides a list of jobs and their status
+pjw get ID        Provides a detailed status of a specific job
+pjw start CMD     Create and start a job; returns the job id
+pjw stop ID       Stop a job. Returns success if the job is already stopped/done
+pjw logs ID       Stream the logs of a job, always starting at the beginning
+```
+
+### Security
+
+#### Authentication
+
+Authentication is based on mTLS and is also completely manual to setup. The server needs a Certificate Authority (CA) and public and private server certificates. It does not generate any of this on it's own. Securely storing and presenting the CA and private server certs is the responsibility of the application operator.
+
+Examples for creating certs:
+
+```shell
+DAYS=20
+SUBJECT="/C=US/ST=Washington/L=Seattle/O=GhostSquad/OU=Prototype Job Worker/CN=localhost"
+
+# CA
+openssl req -newkey rsa:4096 \
+  -new -nodes -x509 \
+  -days "${DAYS}" \
+  -out ca.crt \
+  -keyout ca.key \
+  -subj "${SUBJECT}"
+
+# Server
+openssl genrsa -out server.key 4096
+openssl req -new -key server.key -days "${DAYS}" -out server.csr \
+    -subj "${SUBJECT}"
+
+openssl x509  -req -in server.csr \
+    -extfile <(printf "subjectAltName=DNS:localhost") \
+    -CA ca.crt -CAkey ca.key  \
+    -days "${DAYS}" -sha256 -CAcreateserial \
+    -out server.crt
+
+# Client
+CLIENT=john
+
+openssl genrsa -out "${CLIENT}.key" 4096
+
+openssl req -new -key "${CLIENT}.key" -days "${DAYS}" -out "${CLIENT}.csr" \
+    -subj "${SUBJECT}/clientName=${CLIENT}"
+
+openssl x509  -req -in "${CLIENT}.csr" \
+    -extfile <(printf "subjectAltName=DNS:localhost") \
+    -CA ca.crt -CAkey ca.key -out "${CLIENT}.crt" -days "${DAYS}" -sha256 -CAcreateserial
+```
+
+Out of scope: Improvements to this process including easy to use script/binary for generating client certificates. Generating client certificates from application itself, and having the user download it. Using external certificate management software such as Vault.
+
+All actions from the CLI require providing the client cert as an option (e.g. `pjw --client-crt /path/to/crt --client-key /path/to/key`)
+
+Out of scope: Improvement to authorizing every request would be a "login" command, which generates a short-lived token, and that token is what is validated.
+
+#### Authorization Schema
+
+##### Actions
+
+- list
+- get
+- start
+- stop
+- logs
+
+All actions are scoped to the current user. As an example, that means `list` will only ever return jobs that were started by that user.
```

Comment:
You should spend a few words on what "current user" means and how authentication ends up attributing a user to a given request.